browser/ui/views/brave_actions/brave_actions_container.cc

```diff
@@ -0,0 +1,195 @@
+  extensions::ExtensionSystem::Get(profile)->ready().Post(
+      FROM_HERE, base::Bind(&BraveActionsContainer::OnExtensionSystemReady,
+                            weak_ptr_factory_.GetWeakPtr()));
+}
+
+BraveActionsContainer::~BraveActionsContainer() {}
+
+void BraveActionsContainer::Init() {
+  // automatic layout
+  auto vertical_container_layout = std::make_unique<views::BoxLayout>(
+                                                views::BoxLayout::kHorizontal);
+  vertical_container_layout->set_main_axis_alignment(
+                                  views::BoxLayout::MAIN_AXIS_ALIGNMENT_CENTER);
+  vertical_container_layout->set_cross_axis_alignment(views::BoxLayout::CROSS_AXIS_ALIGNMENT_CENTER);
+  SetLayoutManager(std::move(vertical_container_layout));
+
+  // children
+  views::Separator* brave_button_separator_ = new views::Separator();
+  // TODO: theme color
+  brave_button_separator_->SetColor(SkColorSetRGB(0xb2, 0xb5, 0xb7));
+  brave_button_separator_->SetPreferredSize(gfx::Size(2,
+                            GetLayoutConstant(LOCATION_BAR_ICON_SIZE)));
+  AddChildView(brave_button_separator_);
+}
+
+void BraveActionsContainer::AddShields(
+                                const extensions::Extension* brave_extension) {
+  DCHECK(brave_extension);
+  VLOG(1) << "AddShields, was already loaded:" << (bool)shields_button_view_;
+  if (!shields_button_view_) {
+    // Create a ExtensionActionViewController for the extension
+    // Passing |nullptr| instead of ToolbarActionsBar since we do not require that
+    // logic.
+    // If we do require notifications when popups are open or closed,
+    // then we should inherit and pass |this| through.
+    shields_view_controller_ = new ShieldsActionViewController(brave_extension,
+        browser_, extension_action_manager_->GetExtensionAction(*brave_extension),
+                                                                        nullptr);
+    shields_button_view_ = new ToolbarActionView(
+                                                  shields_view_controller_, this);
+    // Sets overall size of button but not image graphic. We set a large width
+    // in order to give space for the bubble.
+    shields_button_view_->SetPreferredSize(gfx::Size(32, 24));
+    // Add extension view after separator view
+    AddChildViewAt(shields_button_view_, 1);
+    Update();
+  }
+}
+
+void BraveActionsContainer::RemoveShields() {
+  VLOG(1) << "RemoveShields, was already loaded:" << (bool)shields_button_view_;
+  if (shields_button_view_) {
+    RemoveChildView(shields_button_view_);
+    delete shields_button_view_;
+    delete shields_view_controller_;
+    shields_view_controller_ = nullptr;
+    shields_button_view_ = nullptr;
+    Update();
+  }
+}
+
+void BraveActionsContainer::Update() {
+  if (shields_view_controller_)
+    shields_view_controller_->UpdateState();
+  // only show separator if we're showing any buttons
+  const bool visible = !should_hide_ && shields_button_view_;
```

Comment:
How about checking children count() instead of checking |shields_button_view_| pointer?
In the near future, I assume that this view can have more buttons.
Because of this, I think we should not assume this view can have only one button.

Reply:
Interesting idea. My intention was to add the one other button (Brave Rewards) to this conditional, once the button is added. I think checking explicitly what we require to be visible is appropriate since we may have a changing number of visual-only elements, such as the separator.